server.rb

```diff
@@ -103,9 +106,13 @@ def redis
   end
 
   def each(&each_block)
-    # Generate a random name to differentiate clients  Duplicates will break
-    # everything, but ¯\_(ツ)_/¯
+    # Generate a random name to differentiate clients
+    # If a name already exists, the name is rerolled
+    begin
     client_id = Faker::Name.first_name + rand(1000).to_s
+    end while Redis.new(url: ENV['REDIS_URL']).sismember(CLIENT_IDS_KEY, client_id)
+    Redis.new(url: ENV['REDIS_URL']).sadd(CLIENT_IDS_KEY, client_id)
```

Comment:
On both 113 and 114, you can just use the IndexStreamer object's existing `redis` method (like we do down in the `messages_html` method):
```
    begin
      client_id = Faker::Name.first_name + rand(1000).to_s
    end while redis.sismember(CLIENT_IDS_KEY, client_id)
    redis.sadd(CLIENT_IDS_KEY, client_id)
```

Reply:
@kkuchta I'll free some time in my schedule today to work your changes in :+1: